### PR TITLE
Multipart upload - Repo key and repo path encoding 

### PR DESCRIPTION
**Pull request by TamirHadad**
- [ ] All [tests](https://github.com/jfrog/jfrog-client-go#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [ ] All [static analysis checks](https://github.com/jfrog/jfrog-client-go/actions/workflows/analysis.yml) passed.
- [ ] This pull request is on the dev branch.
- [ ] I used gofmt for formatting the code before submitting the pull request.
-----

I have read the CLA Document and I hereby sign the CLA

### CODE REVIEW

**Comment by TamirHadad** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by TamirHadad** [comment]
recheck

File: artifactory/services/utils/multipartupload_test.go [hunk 1]
```diff
@@ -78,30 +78,50 @@ func TestUnsupportedVersion(t *testing.T) {
...
 
-	// Execute CreateMultipartUpload
-	actualToken, err := multipartUpload.createMultipartUpload(repoKey, repoPath, partSize)
-	assert.NoError(t, err)
-	assert.Equal(t, token, actualToken)
+	for _, tt := range testCases {
+		t.Run(tt.name, func(t *testing.T) {
+			handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				println(r.URL.Path)
+				println(r.URL.RawQuery)
```
Comment (yahavi) [inline artifactory/services/utils/multipartupload_test.go]:
```suggestion
```

File: artifactory/services/utils/multipartupload.go [hunk 2]
```diff
@@ -262,8 +263,9 @@ func (mu *MultipartUpload) uploadPart(logMsgPrefix, localPath string, fileSize,
 }
 
 func (mu *MultipartUpload) createMultipartUpload(repoKey, repoPath string, partSize int64) (token string, err error) {
-	url := fmt.Sprintf("%s%screate?repoKey=%s&repoPath=%s&partSizeMB=%d", mu.artifactoryUrl, uploadsApi, repoKey, repoPath, partSize/SizeMiB)
-	resp, body, err := mu.client.SendPost(url, []byte{}, mu.httpClientsDetails)
+	requestUrl := fmt.Sprintf("%s%screate?repoKey=%s&repoPath=%s&partSizeMB=%d", mu.artifactoryUrl, uploadsApi, url.QueryEscape(repoKey), url.QueryEscape(repoPath), partSize/SizeMiB)
+	log.Debug("Request final url", requestUrl)
```
Comment (yahavi) [inline artifactory/services/utils/multipartupload.go]:
```suggestion
```

There is already a debug log in the HTTP client: https://github.com/jfrog/jfrog-client-go/blob/v1.43.1/http/httpclient/client.go#L175

**Review by yahavi — APPROVED** [review]

**Pull request merged**